canonical-web-and-design/vanilla-framework
# Make sticky side navigation optional

## Done

Removes sticky functionality from default side navigation.
Adds `is-sticky` modifier class to enable sticky side navigation on large screens.


Fixes #3206

## QA

- Run `./run` or [demo]( https://vanilla-framework-3427.demos.haus/docs)
- Check the examples of side navigation (both desktop and mobile view) and make sure they work as expected:
  - [default](https://vanilla-framework-3427.demos.haus/docs/examples/patterns/side-navigation/docs) and [with icons](https://vanilla-framework-3427.demos.haus/docs/examples/patterns/side-navigation/icons) or [raw HTML](https://vanilla-framework-3427.demos.haus/docs/examples/patterns/side-navigation/raw-html) - should not be sticky on large screen, mobile drawer should work as before
  - new [sticky](https://vanilla-framework-3427.demos.haus/docs/examples/patterns/side-navigation/sticky) variant should be sticky when page is scrolled. bigger example in [docs layout](https://vanilla-framework-3427.demos.haus/docs/examples/layouts/documentation_
  - verify also comparison examples [light](https://vanilla-framework-3427.demos.haus/docs/examples/patterns/side-navigation/default) and [dark](https://vanilla-framework-3427.demos.haus/docs/examples/patterns/side-navigation/dark) - only sticky nav (3rd) should be sticky
- Check updated [side navigation docs](https://vanilla-framework-3427.demos.haus/docs/patterns/navigation#sticky)
